changes.go

```diff
@@ -490,7 +509,21 @@ func (ch *AddApplicationChange) Description() []string {
 		}
 		unitsInfo = fmt.Sprintf(" with %d unit%s", ch.Params.NumUnits, plural)
 	}
-	return []string{fmt.Sprintf("deploy application %s%s%s using %s", ch.Params.Application, unitsInfo, series, ch.Params.charmURL)}
+
+	var using string
+	var location string
+	curl, err := charm.ParseURL(ch.Params.charmURL)
+	if err == nil {
+		location = storeLocation(curl.Schema)
+		if location != "" {
+			location = fmt.Sprintf(" %s ", location)
+		}
+		if ch.Params.Application != curl.Name {
+			using = fmt.Sprintf(" using %s", curl.Name)
+		}
+	}
+
+	return []string{fmt.Sprintf("deploy%sapplication %s%s%s%s", location, ch.Params.Application, unitsInfo, series, using)}
```

Comment:
Let's make this description closer to what we do for juju deploy:

"deploy application mariadb from charm-store with 2 units on kubernetes"